### PR TITLE
feature/CORE-582/fix-js-key-for-duplicated-block

### PR DESCRIPTION
**Pull request by bulatgab**
See main PR for demo, etc.
https://bitbucket.org/voto/voto5/pull-requests/3350

### CODE REVIEW

File: src/components/interaction-designer/toolbar/TreeBuilderToolbar.vue [hunk 1]
```diff
@@ -234,12 +234,15 @@
...
-                   @click.prevent="handleAddBlockByTypeSelected(classDetails)">
+              <li
+                v-if="Object.keys(blockClassesForBranchingCategory).length === 1"
+                class="nav-item">
+                <a
+                  v-for="(classDetails, className) in blockClassesForBranchingCategory"
+                  :key="className + 'item'"
+                  class="nav-link single-menu"
+                  href="#"
+                  @click.prevent="handleAddBlockByTypeSelected(classDetails)">
```
Comment (bulatgab) [inline src/components/interaction-designer/toolbar/TreeBuilderToolbar.vue:245]: _ESLint fixes_

File: src/components/interaction-designer/toolbar/TreeBuilderToolbar.vue [hunk 2]
```diff
@@ -314,12 +317,15 @@
         </div>
       </div>
     </div>
-    <div v-if="isBuilderCanvasEnabled" class="tree-builder-toolbar-alerts w-100">
+    <div
+      v-if="isBuilderCanvasEnabled"
+      class="tree-builder-toolbar-alerts w-100">
```
Comment (bulatgab) [inline src/components/interaction-designer/toolbar/TreeBuilderToolbar.vue:322]: _ESLint fixes_

File: src/store/flow/flow.ts [hunk 3]
```diff
@@ -65,6 +65,8 @@ export const getters: GetterTree<IFlowsState, IRootState> = {
       } catch (err) {
         return undefined
       }
+    } else {
+      return undefined
```
Comment (bulatgab) [inline src/store/flow/flow.ts:69]: _ESLint fix_

File: src/store/flow/flow.ts [hunk 4]
```diff
@@ -112,7 +114,7 @@ export const getters: GetterTree<IFlowsState, IRootState> = {
 
 export const mutations: MutationTree<IFlowsState> = {
   flow_resetStates(state: IFlowsState) {
-    Object.assign(state, stateFactory() as IFlowsState)
+    Object.assign(state, stateFactory())
```
Comment (bulatgab) [inline src/store/flow/flow.ts:117]: _ESLint fix_

File: src/store/flow/flow.ts [hunk 5]
```diff
@@ -579,13 +581,17 @@ export const actions: ActionTree<IFlowsState, IRootState> = {
     state.selectedBlocks = []
   },
 
-  async flow_duplicateAllSelectedBlocks({state, dispatch}) {
-    const newBlocksUuid: string[] = []
-    forEach(state.selectedBlocks, async (blockId: IBlock['uuid']) => {
-      const duplicatedBlock: IBlock = await dispatch('flow_duplicateBlock', {blockId})
-      newBlocksUuid.push(duplicatedBlock.uuid)
-    })
-    state.selectedBlocks = newBlocksUuid
```
Comment (bulatgab) [inline src/store/flow/flow.ts:588]: This code was not awaiting for individual `flow_duplicateBlock` actions.
So returning `newBlocksUuid` from this function was giving an empty (at least initially) array.

File: src/components/interaction-designer/block/BlockToolbar.vue [hunk 6]
```diff
@@ -110,14 +109,3 @@ export class BlockToolbar extends mixins(Lang) {
 }
 export default BlockToolbar
 </script>
-
-<style scoped>
```
Comment (safydy-r) [inline src/components/interaction-designer/block/BlockToolbar.vue:114]: yes, unused

**Review by safydy-r — APPROVED** [review]
looks good to me, thanks @bulatgab

**Pull request merged**